lib/physicians/resolvers.ts

```diff
@@ -0,0 +1,75 @@
+import { ApolloError } from 'apollo-server-errors'; // Apollo error
+import { Resolver } from '@lib/resolvers'; // Resolver type
+import { DBErrorCode } from '@lib/db/errors'; // Database errors
+import { PhysicianError } from './errors'; // Physician error
+import { Prisma } from '.prisma/client';
+
+/**
+ * Query all physicians of RCD applicants
+ * @returns All physicians of RCD applicants
+ */
+export const physicians: Resolver = async (_parent, _args, { prisma }) => {
+  const physicians = await prisma.physician.findMany();
+  return physicians;
+};
+
+/**
+ * Create a physician
+ * @returns Status of operation (ok, error)
+ */
+export const createPhysician: Resolver = async (_, args, { prisma }) => {
+  const {
+    input: {
+      firstName,
+      lastName,
+      mspNumber,
+      address,
+      city,
+      province,
+      postalCode,
+      phone,
+      status,
+      notes,
+    },
+  } = args;
+
+  let physician;
+  try {
+    physician = await prisma.physician.create({
+      data: {
+        firstName,
+        lastName,
+        mspNumber,
+        address,
+        city,
+        province,
+        postalCode,
+        phone,
+        status,
+        notes,
+      },
+    });
+  } catch (err) {
+    if (err instanceof Prisma.PrismaClientKnownRequestError) {
+      if (err.code === DBErrorCode.UniqueConstraintFailed) {
+        throw new PhysicianError(
+          'Unique constraint failed, a new physician cannot be created with this MSP number'
+        );
+      }
+      if (err.code === DBErrorCode.LengthConstraintFailed) {
+        throw new PhysicianError(
+          "Length constraint failed, provided value too long for the column's type"
+        );
```

Comment:
I handled each error just by code and not by the column since the [LengthConstraintError](https://www.prisma.io/docs/reference/api-reference/error-reference#p2000) didn't actually return which column was failing in the `meta `property for some reason. 
![image](https://user-images.githubusercontent.com/55164714/121789267-94694700-cba2-11eb-8880-ebd29aadfd75.png)

Also, there are a lot of fields that have this length constraint on them so making an error object + adding an extra conditional for each one would start getting pretty ugly imo. 

Don't feel too strongly about this tho so definitely open to other ideas

Reply:
I see. Sounds good. In the future, let's investigate and see if we can include the field that failed the length constraint in the error message.